sdk/core/azure-core/tests/testserver_tests/test_rest_headers.py

```diff
@@ -4,154 +4,247 @@
+    assert list(h) == ['a', 'b']
+
+    assert list(h.keys()) == ["a", "b"]
+    assert list(h.values()) == ["123, 456", "789"]
+    assert list(h.items()) == [("a", "123, 456"), ("b", "789")]
+    assert list(h) == ["a", "b"]
+    assert dict(h) == {"a": "123, 456", "b": "789"}
+
+def test_headers_response_keys(get_response_headers):
+    h = get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    # basically want to make sure this behaves like dict {"a": "123, 456", "b": "789"}
+    ref_dict = {"a": "123, 456", "b": "789"}
+    assert list(h.keys()) == list(ref_dict.keys())
+    if sys.version_info < (3, 0):
+        assert repr(h.keys()) == "['a', 'b']"
     else:
-        assert set(list(h.values())) == set(["a", "123", "c"])
-
-
-def test_headers_insert_appends_if_new():
-    h = _get_headers({"a": "a", "b": "b", "c": "c"})
-    h["d"] = "123"
-    if sys.version_info >= (3, 6):
-        assert list(h.values()) == ["a", "b", "c", "123"]
+        assert repr(h.keys()) == "KeysView({'a': '123, 456', 'b': '789'})"
+    assert "a" in h.keys()
+    assert "b" in h.keys()
+    assert set(h.keys()) == set(ref_dict.keys())
+
+@pytest.mark.skipif(sys.version_info < (3, 0),
+                    reason="In 2.7, .keys() are not mutable")
+def test_headers_response_keys_mutability(get_response_headers):
+    h = get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    # test mutability
+    before_mutation_keys = h.keys()
+    h['c'] = '000'
+    assert 'c' in before_mutation_keys
+
+def test_headers_response_values(get_response_headers):
+    h = get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    # basically want to make sure this behaves like dict {"a": "123, 456", "b": "789"}
+    ref_dict = {"a": "123, 456", "b": "789"}
+    assert list(h.values()) == list(ref_dict.values())
+    if sys.version_info < (3, 0):
+        assert repr(h.values()) == "['123, 456', '789']"
     else:
-        assert set(list(h.values())) == set(["a", "b", "c", "123"])
-
-
-def test_headers_insert_removes_all_existing():
-    h = _get_headers([("a", "123"), ("a", "456")])
-    h["a"] = "789"
-    assert dict(h) == {"a": "789"}
-
-
-def test_headers_delete_removes_all_existing():
-    h = _get_headers([("a", "123"), ("a", "456")])
-    del h["a"]
-    assert dict(h) == {}
+        assert repr(h.values()) == "ValuesView({'a': '123, 456', 'b': '789'})"
+    assert '123, 456' in h.values()
+    assert '789' in h.values()
+    assert set(h.values()) == set(ref_dict.values())
+
+@pytest.mark.skipif(sys.version_info < (3, 0),
+                    reason="In 2.7, .values() are not mutable")
+def test_headers_response_values_mutability(get_response_headers):
+    h = get_response_headers(HttpRequest("GET", "/headers/duplicate/numbers"))
+    # test mutability
+    before_mutation_values = h.values()
+    h['c'] = '000'
+    assert '000' in before_mutation_values
+
+def test_headers_response_items(get_response_headers):
```

Comment:
in 2.7, `collections.KeysView`, `collections.ItemsView`, and `collections.ValuesView` are immutable, while in `collections.abc`, they are mutable:

2.7 code snippet 
<img width="261" alt="Screen Shot 2021-08-23 at 12 09 02 PM" src="https://user-images.githubusercontent.com/43154838/130480738-bc99472f-5c42-48be-9cd3-7fee025a5ab4.png">

code snippet in 3.9:
<img width="256" alt="Screen Shot 2021-08-23 at 12 09 50 PM" src="https://user-images.githubusercontent.com/43154838/130480849-778fb92d-2854-4d5b-bb4b-986cb20f9c8c.png">